core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -364,7 +368,10 @@ public long pageSizeBytes() {
    *
    * @throws TooLargePageException
    */
-  public MemoryBlock allocatePage(long size, MemoryConsumer consumer) {
+  public MemoryBlock allocatePage(
+      long size,
+      MemoryConsumer consumer,
+      boolean isRetry) {
```

Comment:
Add. this avoid print too much stack information.

Reply:
given we are touching this place, can we use `int retryCount` instead of a boolean? And we can also log the retry count.

Reply:
Done, how about current code.